adapters/httpapi_compact.c

```diff
@@ -744,6 +746,12 @@ static HTTPAPI_RESULT OpenXIOConnection(HTTP_HANDLE_DATA* http_instance)
             result = HTTPAPI_SET_OPTION_FAILED;
             LogInfo("Could not load the client certificate private key");
         }
+        else if ((http_instance->tls_renegotiation != 0) &&
+            (xio_setoption(http_instance->xio_handle, OPTION_SET_TLS_RENEGOTIATION, &http_instance->tls_renegotiation) != 0))
```

Comment:
I checked the tlsio adapters and only this actually use this setting: [tlsio_mbedtls.c](https://github.com/Azure/azure-c-shared-utility/blob/98666e9b4d46e63d900d4abd1bc1591d3049b5a6/adapters/tlsio_mbedtls.c#L1026)

And in that implementation, the expected data type is `bool`. Could you please update `tls_negotiation` to be `bool` ?

Reply:
Fixed on the commit 98666e9b4d46e63d900d4abd1bc1591d3049b5a6